.circleci/config.yml

```diff
@@ -0,0 +1,145 @@
+# These environment variables must be set in CircleCI UI
+#
+# NPM_TOKEN - A valid NPM token for releases
+#
+# NOTE:
+# - to validate changes to this file locally using the circleci CLI tool:
+#
+#     circleci config process .circleci/config.yml
+#
+# - to try run jobs locally:
+#
+#     circleci config process .circleci/config.yml > tmp/processed.yaml
+#     circleci local execute -c tmp/processed.yml --job build-nodejs-current
+#
+version: 2.1
+
+references:
+  # NOTE: update the nodejs version strings that follows to change the
+  # set of nodejs versions that should be tested.
+  # The nodejs version set as `nodejs_current` is the one used to
+  # release the package on npm.
+  nodejs_versions:
+    - &nodejs_current "10"
+    - &nodejs_next "12"
+    - &nodejs_experimental "14"
+
+  nodejs_enum: &nodejs_enum
+    type: enum
+    default: *nodejs_current
+    enum:
+      - *nodejs_current
+      - *nodejs_next
+      - *nodejs_experimental
+  repo_path: &repo_path ~/web-ext
+  defaults: &defaults
+    working_directory: *repo_path
+
+commands:
+  attach_project_repo:
+    description: attach repo from workspace
+    steps:
+      - attach_workspace:
+          at: *repo_path
+
+  persist_project_repo:
+    description: persist repo in workspace
+    steps:
+      - persist_to_workspace:
+          root: *repo_path
+          paths: .
+
+  restore_build_cache:
+    description: restore npm package cache
+    parameters:
+      suffix:
+        type: string
+        default: default
+    steps:
+      - restore_cache:
+          keys:
+            - npm-packages-{{ checksum "package-lock.json" }}-<< parameters.suffix >>
```

Comment:
This is cool, we could do this more everywhere else to save on duplication on cache creation/restoration.

Reply:
yep, it is, the jobs do definitely run faster thanks to it and the checksum and suffix clear my concerns about stale caches and cached node_modules dirs built with different npm major versions.

If I'm not mistaken we are already using this trick in all the repository where we do have a lock file (and I'm 100% sure that we use it for the addons-linter and web-ext repositories).